src/Hamiltonians/reduced_density_matrix.jl

```diff
@@ -121,3 +121,84 @@ function get_offdiagonal(
     address, value = excitation(addr, dst, src)
     return address, value
 end
+
+"""
+    ReducedDensityMatrix(addr::SingleComponentFockAddress; n = 1) <: AbstractOperator
+
+Represent the n-particle reduced density matrix:
+
+```math
+ρ̂^{(n)}_{j_1,...,j_1,k_1,...,k_n} =  \\prod_{i}^{n} â^†_{j_i} \\prod_{l}^{n} â_{k_{n+1-l}}
+```
+
+Where `j_i` and `k_i` (all `<: Int`) specify the single particle sites on a lattice.
+Additionally, the indices run in the following manners:
+
+```math
+j_n> ... > j_{i+1} > j_{i} > ... > j_1 and k_n> ... > k_{i+1} > k_{i} > ... > k_1
+```
```

Comment:
There are typos in the LaTeX formulas. Also I don't really understand what is written. Can you express this more clearly? Provide information that a potential user would need to know. What is this good for? How do you use it?

Maybe giving an example for how this can be used (and the full reduced density matrix be restored, eg. for use in a diagonalisation function) would be helpful.

Reply:
I made some changes to the Doc. Please have a look.